src/java/org/apache/cassandra/locator/Replicas.java

```diff
@@ -50,6 +50,30 @@
     public abstract int size();
     protected abstract Collection<Replica> getUnmodifiableCollection();
 
+
+    public boolean equals(Object o)
+    {
+        if (this == o) return true;
+        if (!(o instanceof Replicas))
+            return false;
+
+        Replicas that = (Replicas) o;
+        if (this.size() != that.size())
+            return false;
+        return Iterables.elementsEqual(this, that);
+    }
+
+
+    public int hashCode()
+    {
+        int result = 17;
```

Comment:
Result starts as 17? I don't recall seeing that being done normally.

Reply:
that's how it's done in Effective Java, bottom of page 48